PolyclinicLab/Polyclinic.Api.Host/Controllers/AnalyticsController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Polyclinic.Application.Contracts;
+using Polyclinic.Application.Interfaces;
+
+namespace Polyclinic.Api.Host.Controllers;
+
+/// <summary>
+/// Provides analytical endpoints that work with live in-memory data (same as CRUD operations).
+/// </summary>
+[ApiController]
+[Route("api/[controller]")]
+public class AnalyticsController(IAnalyticsService analyticsService, ILogger<AnalyticsController> logger) : ControllerBase
+{
+    /// <summary>
+    /// (1) Returns all doctors with at least 10 years of experience.
+    /// </summary>
+    [HttpGet("experienced-doctors")]
+    public ActionResult<List<DoctorDto>> GetExperiencedDoctor([FromQuery] int minExperience = 10)
+    {
+        try
+        {
+            if (minExperience < 0 || minExperience > 100)
+            {
+                logger.LogWarning("Invalid minExperience value: {MinExperience}", minExperience);
+                return BadRequest("Experience value must be between 0 and 100.");
+            }
```

Comment:
Можно было сделать проще воспользовавшись Range атрибутом:

```suggestion
    public ActionResult<List<DoctorDto>> GetExperiencedDoctor([FromQuery][Range(0, 100)] int minExperience = 10)
    {
        try
        {
```
Можете оставить свой вариант, я не против. Просто пишу что имеется такой способ валидации данных.